### PR TITLE
Add delete option to allow keys removal

### PR DESCRIPTION
**Pull request by lukaszkowalczyk98**
Requestor/Issue: Lukasz Kowalczyk
Risk (low/med/high): low/med
Tested (yes/no):yes
Description/Why: We would like to extend treasury with the removal keys option. Thanks to delete flag we would be able to delete specific keys from our treasury.

### CODE REVIEW

File: client/delete_test.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+		S3BucketName: "fake_s3_bucket",
+	}
+	treasury, err := client.New(dummyClientOptions)
+	if err != nil {
+		t.Error(err)
+	}
+
+	err = treasury.Delete(test.Key1)
+	if err != nil {
+		t.Error(err)
```
Comment (jadrol) [inline client/delete_test.go:23]: what are we testing here?

File: cmd/delete.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	"fmt"
+
+	"github.com/AirHelp/treasury/client"
+	"github.com/spf13/cobra"
+)
+
+// deleteCmd represents the delete command
+var deleteCmd = &cobra.Command{
+	Use:   "delete KEY",
+	Short: "Remove secrets from Treasury",
```
Comment (jadrol) [inline cmd/delete.go:28]:
```suggestion
	Short: "Remove secret from Treasury",
```

File: README.md [hunk 3]
```diff
@@ -151,6 +152,14 @@ development/application/app_api_pass
...
 ```
 
+### Delete secret
+Delete the secret from treasury
+
+```
+> treasury delete development/application/secret_key
+Key development/application/secret_key has been successfully deleted
+```
+
```
Comment (jadrol) [inline README.md:162]: below in Readme you have examples of policies for S3 and SSM. You need to also add all needed there for handling `delete`

File: client/delete_test.go (hunk 1 above)
Comment (lukaszkowalczyk98) [reply client/delete_test.go:23]: I've tried to test whether the function works when there is existing key in treasury.

File: README.md (hunk 3 above)
Comment (lukaszkowalczyk98) [reply README.md:162]: sure thing added

**Review by jadrol — APPROVED** [review]

**Pull request merged**